basic_network_test.go

```diff
@@ -1138,53 +1133,17 @@ func TestMasterPlaylistIntegration(t *testing.T) {
 	if err != nil {
 		t.Errorf("Error getting master playlist: %v", err)
 	}
-	timer := time.NewTimer(time.Second * 3)
 	select {
 	case r := <-mplc:
 		vars := r.Variants
 		if len(vars) != 1 {
 			t.Errorf("Expecting 1 variants, but got: %v - %v", len(vars), r)
 		}
-	case <-timer.C:
+	case <-time.After(time.Second * 3):
 		glog.Infof("n2 mplMap: %v", n2.mplMap)
 		t.Errorf("Timed out")
 	}
 
```

Comment:
Removing this test case because it's not really related to the MasterPlaylist integration test (more of a general networking thing).  We should rethink how to deal with / test flaky nodes.

Reply:
> We should rethink how to deal with / test flaky nodes.

Could we stub out the actual networking calls such that they fail in a way that the test prescribes? That way we can check how the connected components (or nodes) respond. Not sure how conducive the current interfaces are to doing that.

Reply:
I think ideally, we should stub out the networking calls.  Theoretically a lot of the "unit tests" here are really integration test.  But as you said, the code is not written in a way that's easily stub-able.  May need some structural changes to accommodate for that.